src/Faker/Provider/en_US/PhoneNumber.php

```diff
@@ -4,30 +4,118 @@
+        '{{areaCode}}.{{exchangeCode}}.####',
+
+        '{{areaCode}}-{{exchangeCode}}-####',
+        '({{areaCode}}) {{exchangeCode}}-####',
+        '1-{{areaCode}}-{{exchangeCode}}-####',
+        '{{areaCode}}.{{exchangeCode}}.####',
+
+        // Extensions
+        '{{areaCode}}-{{exchangeCode}}-#### x###',
+        '({{areaCode}}) {{exchangeCode}}-#### x###',
+        '1-{{areaCode}}-{{exchangeCode}}-#### x###',
+        '{{areaCode}}.{{exchangeCode}}.#### x###',
+
+        '{{areaCode}}-{{exchangeCode}}-#### x####',
+        '({{areaCode}}) {{exchangeCode}}-#### x####',
+        '1-{{areaCode}}-{{exchangeCode}}-#### x####',
+        '{{areaCode}}.{{exchangeCode}}.#### x####',
+
+        '{{areaCode}}-{{exchangeCode}}-#### x#####',
+        '({{areaCode}}) {{exchangeCode}}-#### x#####',
+        '1-{{areaCode}}-{{exchangeCode}}-#### x#####',
+        '{{areaCode}}.{{exchangeCode}}.#### x#####'
     );
+
+    /**
+     * @see https://en.wikipedia.org/wiki/Toll-free_telephone_number#United_States
+     */
+    protected static $tollFreeAreaCodes = array(
+        800, 844, 855, 866, 877, 888
+    );
+    protected static $tollFreeFormats = array(
+        // Standard formats
+        '{{areaCode}}-{{exchangeCode}}-####',
+        '({{areaCode}}) {{exchangeCode}}-####',
+        '1-{{areaCode}}-{{exchangeCode}}-####',
+        '{{areaCode}}.{{exchangeCode}}.####',
+    );
+
+    public function tollFreePhoneNumber()
+    {
+        $areaCode = self::randomElement(static::$tollFreeAreaCodes);
+        $format = self::randomElement(static::$tollFreeFormats);
+
+        $format = str_replace('{{areaCode}}', $areaCode, $format);
+
+        return self::numerify($this->generator->parse($format));
+    }
+
+    /**
+     * NPA-format area code
+     *
+     * @see https://en.wikipedia.org/wiki/North_American_Numbering_Plan#Numbering_system
+     *
+     * @return string
+     */
+    public static function areaCode()
+    {
+        $digits[] = self::numberBetween(2, 9);
+        $digits[] = self::randomDigit();
+        $digits[] = self::randomDigitNot($digits[1]);
+
+        return join('', $digits);
+    }
+
+    /**
+     * NXX-format central office exchange code
+     *
+     * @see https://en.wikipedia.org/wiki/North_American_Numbering_Plan#Numbering_system
+     *
+     * @return string
+     */
+    public static function exchangeCode()
+    {
+        $digits[] = self::numberBetween(2, 9);
+        $digits[] = self::randomDigit();
+
+        if ($digits[1] === 1) {
+            $digits[] = self::randomDigitNot(1);
+        } else {
+            $digits[] = self::randomDigit();
+        }
+
+        return join('', $digits);
+    }
+
+    /**
+     * Generates a random digit, which cannot be $except
+     *
+     * @param int $except
+     * @return int
+     */
+    private static function randomDigitNot($except)
+    {
+        $result = self::numberBetween(0, 8);
+        if ($result >= $except) {
```

Comment:
No, to make it random among the 9 possibilities, it shifts the whole range over the selected nimber up. For example, if 5 is passed as the exception, it generates a digit from 0-8 then shifts anything over 5 up one to make the range 0-4,6-9.